Add optional parameter + Fixes

dpoeschl opened the pull request.

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/Parameter.cs:
```diff
@@ -25,22 +25,57 @@ public ExistingParameter(IParameterSymbol param)
+            TypeBinds = typeBinds;
+            TypeName = typeName;
+            Name = name;
             CallSiteValue = callSiteValue;
```
Assignment is unnecessary now.

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/Parameter.cs:
```diff
@@ -25,22 +25,57 @@ public ExistingParameter(IParameterSymbol param)
             Type = type;
-            TypeNameDisplayWithErrorIndicator = typeNameDisplayWithErrorIndicator;
-            ParameterName = parameter;
+            TypeBinds = typeBinds;
```
How is this distinct from whether Type is an error type?

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/Parameter.cs:
```diff
@@ -25,22 +25,57 @@ public ExistingParameter(IParameterSymbol param)
-            TypeNameDisplayWithErrorIndicator = typeNameDisplayWithErrorIndicator;
-            ParameterName = parameter;
+            TypeBinds = typeBinds;
+            TypeName = typeName;
```
How is this distinct from type.Name and friends?

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/IChangeSignatureOptionsService.cs:
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.CodeAnalysis.ChangeSignature
     {
         /// <summary>
-        /// Changes signature of the symbol (currently a method symbol or an event symbol)
+        /// Gets options and produces a <see cref="SignatureChange"/> if successful.
```
```suggestion
        /// Gets options and produces a <see cref="ChangeSignatureOptionsResult"/> if successful.
```

?

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/ChangeSignatureResult.cs:
```diff
@@ -10,10 +10,14 @@ internal sealed class ChangeSignatureResult
         public bool PreviewChanges { get; }
 
+        /// <summary>
+        /// Name of the symbol. Needed here for the Preview Changes dialog.
```
What is "the symbol"?

jasonmalinowski commented on src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs:
```diff
@@ -36,28 +36,29 @@ public AddParameterDialogViewModel(Document document, int insertPosition)
             }
         }
 
+        public bool TypeBinds => !TypeSymbol!.IsErrorType();
```
```suggestion
        public bool TypeBinds => !TypeSymbol.IsErrorType();
```

IsErrorType is an extension method.

jasonmalinowski commented on src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialog.xaml.cs:
```diff
@@ -143,9 +143,13 @@ private void Add_Click(object sender, RoutedEventArgs e)
+                    addParameterViewModel.UseNamedArguments,
+                    addParameterViewModel.IsCallsiteOmitted,
+                    addParameterViewModel.IsCallsiteError,
+                    addParameterViewModel.TypeBinds);
```
Sould we use named parameters here? This is a lot of bools that could easily be mixed up here...

jasonmalinowski commented on src/Features/Core/Portable/ChangeSignature/Parameter.cs:
```diff
@@ -25,22 +25,57 @@ public ExistingParameter(IParameterSymbol param)
+        public string DefaultValue { get; }
+        public bool UseNamedArguments { get; }
+        public bool IsCallsiteOmitted { get; }
+        public bool IsCallsiteError { get; }
```
Document?

sharwell committed:
Fix RS1030 (Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer)

sharwell committed:
Suppress RS1030 where no other option is available

sharwell committed:
Update Microsoft.CodeAnalysis.Testing reference

sharwell committed:
Simplify structure tests using new markup syntax

cston committed:
Remove IEnumerator<T> from ArrayBuilder.Enumerator (#43121)

Commit:
Fix formatting and enable nullable

sharwell committed:
Merge pull request #43095 from sharwell/use-cache

🐇 Use config cache to avoid tree creation

sharwell committed:
Merge pull request #43077 from sharwell/batch-nested-fixes

Update BatchFixAllProvider to support nested code actions

Commit:
Merge pull request #43115 from dibarbet/merge_167_to_master

Merge release/dev16.7-preview1 to master

mavasani committed:
Move extension method to shared analyzer layer

mavasani committed:
Port C# InlineDeclaration analyzer/fixer/tests to shared layer

mavasani committed:
xlf and resx file changes

mavasani committed:
Merge remote-tracking branch 'upstream/master' into PortSimplifyConditional

sharwell committed:
Merge pull request #43100 from sharwell/is-match

🐇 Use Regex.IsMatch instead of Regex.Match when no groups are needed

Commit:
Merge pull request #43135 from mavasani/PortCSharpInlineDeclaration

Port C# InlineDeclaration analyzer/fixer/tests to shared layer

agocke committed:
Add guard in the UnassignedFieldsWalker for a property without a getter (#43127)

BertanAygun committed:
Integrate RichCodeNavigation targets to help with LSIF generation on builds
with indexing environment variable is set.

dpoeschl committed:
Merge remote-tracking branch 'dotnet/release/dev16.7-preview1' into AddOptionalParameter

sharwell committed:
Merge pull request #43102 from sharwell/wait-longer

Prefer rare false negatives in testing to false positives

sharwell committed:
Add tests for parameter lists

Commit:
Merge pull request #43132 from dotnet/merges/release/dev16.6-to-master

Merge release/dev16.6 to master

Commit:
Respond to CR

JoeRobich committed:
Handle RS1030 diagnostic

mavasani committed:
Add more specific asserts

mavasani committed:
Enable IDE0005 (RemoveUnnecessaryUsings) as build warning in IDE layers (CodeStyle, Features and Workspaces)

dotnet-bot committed:
Apply FixAll in solution for IDE0005 (fixes C# projects in the solution)

genlu committed:
Merge pull request #43072 from genlu/AsyncSuggestion

Suggest interface types when member is implemented as async

JoeRobich committed:
Remove suppression of RS1030

dotnet-bot committed:
Apply FixAll in solution for IDE0005 (fixes VB projects in the solution)

mavasani committed:
Disable the temporary workaround added in .editorconfig to enable testing FixAll in solution operation

dibarbet committed:
Merge pull request #43134 from uniqueiniquity/peekNullException

Handle null IGoToDefinitionService in peek

Commit:
Do not incorrectly "optimize" lifted bool "or" and "and" (#43122)

Fixes #41760
* Skip broken test
Relates to #41456

agocke committed:
Add guard in the UnassignedFieldsWalker for a property without a getter (#43127) (#43146)

(cherry picked from commit 9bdb18aec903d4a628f107f749d31e296770c9a1)

mavasani committed:
Move InferredMemberName simplification helpers to shared layer

mavasani committed:
Port UseInferredMemberName analyzer/fixer/tests to shared layer

mavasani committed:
xlf and resx file changes

BertanAygun committed:
Integrate review feedback to move versions to version.props

mavasani committed:
Port InvokeDelegateWithConditionalAccess to shared layer

mavasani committed:
xlf and resx file changes

dibarbet committed:
Update versions for 16.7

mavasani committed:
Fix conditionally unused usings in presence of `#if CODE_STYLE` for code in shared layer

dibarbet committed:
Update publishing for 16.6 p3, 16.7

mavasani committed:
Merge pull request #43110 from mavasani/PortSimplifyConditional

Port SimplifyConditional analyzer/fixer/tests to shared layer

JoeRobich committed:
Suppress RS2008 for .cs and .vb files

mavasani committed:
Fix conditionally used usings for netstandard2.0 target framework

mavasani committed:
Remove unused usings reported on command line build, but not in the IDE

mavasani committed:
Fix release build - conditionally exclude usings only used for DEBUG only code

mavasani committed:
Merge remote-tracking branch 'upstream/master' into EnforceUnnecessaryUsings

sharwell committed:
Fix Make Field Readonly handling of ref return values

Fixes #33009

Commit:
Merge pull request #43004 from dpoeschl/FixLambdaTypeParameterConcreteizing

Find concrete types for type parameters in completion off of lambda parameters

mmitche committed:
Feed cleanup (#42739)

* Feed cleanup

Remove blob feeds (and some other feeds I don't think are required)

mavasani committed:
Update src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/DocumentBasedFixAllProvider.cs

Co-Authored-By: Sam Harwell <sam@tunnelvisionlabs.com>

Commit:
Merge remote-tracking branch 'dotnet/master' into update-ttn

dibarbet committed:
Don't publish assets to myget during PR validation

Commit:
Merge pull request #43156 from mavasani/PortUseInferredMemberName

Port UseInferredMemberName analyzer/fixer/tests to shared layer

genlu committed:
Add test

genlu committed:
Preserve added imports

mavasani committed:
Fix build from merge

mavasani committed:
Port CSharpUseLocalFunction analyzer/tests to shared layer

Porting the code fix is blocked on #43056

mavasani committed:
xlf and resx file changes

Commit:
Merge pull request #43160 from dotnet/dibarbet-patch-1

Update versions for 16.7

Commit:
Merge pull request #43158 from mavasani/PortInvokeDelegateWithConditionalAccess

Port InvokeDelegateWithConditionalAccess to shared layer

BertanAygun committed:
Merge master branch

BertanAygun committed:
Fix the incorrect merge

Commit:
Merge pull request #43161 from dotnet/dibarbet-patch-2

Update publishing for 16.6 p3, 16.7

mavasani committed:
Merge remote-tracking branch 'upstream/master' into EnforceUnnecessaryUsings

mavasani committed:
Remove unused usings from a newly added file

jcouv committed:
Merge pull request #43169 from jcouv/update-ttn

Merge "target-typed new" feature in 16.7p1

JoeRobich committed:
Merge pull request #43006 from JoeRobich/set-package-version

Set Microsoft.CodeAnalysis.Analyzers package to a release version

sharwell committed:
Merge pull request #36272 from sharwell/track-ref-returns

Track ref returns for Make Field Readonly

BertanAygun committed:
Merge remote-tracking branch 'origin/master' into includeRichNavTarget

genlu committed:
Add argument name

JoeRobich committed:
Merge remote-tracking branch 'origin/release/dev16.6' into merges/release/dev16.5-to-release/dev16.6

AArnott committed:
Merge pull request #43023 from AArnott/nrtFixes

Nullable ref type annotation fixes to analyzer APIs

JoeRobich committed:
Update color scheme on package initialization.

Removes the delay in applying a color scheme on startup.

dibarbet committed:
Merge pull request #43171 from dotnet/dev/dibarbet/dont_publish

Don't publish assets to myget during PR validation

JoeRobich committed:
Add comment about why we update color scheme on initialize

JoeRobich committed:
Merge pull request #43193 from dotnet/merges/release/dev16.5-to-release/dev16.6

Merge release/dev16.5 to release/dev16.6

cston committed:
Reduce stack frame size of CheckConstraints (#43094)

mavasani committed:
Merge pull request #43166 from mavasani/EnforceUnnecessaryUsings

Enforce IDE0005 (remove unnecessary usings) as build warning in few IDE layers

mavasani committed:
Merge pull request #43129 from mavasani/FixAssert

Fix assert in AnalyzerDriver.GetOperationsToAnalyze

JoeRobich committed:
Merge pull request #43200 from dotnet/dev/jorobich/apply-colors-on-initialize

Update color scheme on package initialization.

jasonmalinowski committed:
Fix race where updating an output path and removing a project could throw

We had a race where if you were adding a project's output path, that
could potentially convert metadata references in other projects to
project references, and in the process we would stop watching for file
changes any of the converted references. If at the same time you removed
the project that contained the reference being converted, we might
end up stopping watching file watches twice, and the second would throw.

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/957715

sharwell committed:
Restrict whitespace collapse between methods to Metadata As Source

Commit:
Update feature status. (#43175)

cston committed:
Avoid using SymbolDisplay while binding type (#43181)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20200408.8 (#43220)

- Microsoft.DotNet.Arcade.Sdk: 1.0.0-beta.20202.9 -> 1.0.0-beta.20208.8

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Commit:
Merge pull request #43149 from dotnet/merges/release/dev16.6-to-master

Merge release/dev16.6 to master

jasonmalinowski committed:
Merge pull request #43142 from BertanAygun/includeRichNavTarget

Integrate RichCodeNavigation targets to help with LSIF generation

jmarolf committed:
use static graph to reduce restore times

tmat committed:
Add CompilationOutputFilePath to project snapshot (#41745)

tmat committed:
Stop generating InteractiveComponents VSIX (#43208)

tmat committed:
Add RazorDocumentInfoFactory (#43226)

Commit:
Merge pull request #43230 from dotnet/merges/release/dev16.6-to-master

Merge release/dev16.6 to master

AlekseyTs committed:
Adjust nullability of a target type for an argument of a lifted binary operator. (#43165)

Fixes #39361.

dpoeschl committed:
Merge remote-tracking branch 'dotnet/master' into AddOptionalParameter

tmat committed:
Remove unused using

tmat committed:
Update EditAndContinueWorkspaceServiceFactory.cs

jasonmalinowski committed:
Mark RichCodeNav.EnvVarDump as private asset

Otherwise this package becomes a dependency of all our NuGet packages.

dpoeschl committed:
Remove unnecessary using

sharwell committed:
Updated documentation for formatter

jmarolf committed:
move to build.ps1

sharwell committed:
Enable nullable reference types for the formatter

genlu committed:
Merge pull request #43173 from genlu/FixOverrideCompletion

Fix override completion to preserve added imports

jasonmalinowski committed:
Merge pull request #43210 from jasonmalinowski/fix-metadata-file-watcher-race

Fix race where updating an output path and removing a project could throw

jasonmalinowski committed:
Merge pull request #43236 from dotnet/dev/tmat/RemoveUsing

Remove unused using

sharwell committed:
Merge pull request #43212 from sharwell/normal-outline

Restrict whitespace collapse between methods to Metadata As Source

jmarolf committed:
only restore if using msbuild since the msbuild with .NET Core 3.1 does not have this feature

JoeRobich committed:
Enable SDL Validation in official builds

tmat committed:
Analyzer file reference (#42967)

* Run AnalyzerFileReference tests on Core

* Require absolute path

* Equality

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/roslyn build 20200410.7 (#43254)

- Microsoft.Net.Compilers.Toolset: 3.6.0-3.20170.19 -> 3.7.0-1.20210.7

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

jaredpar committed:
Skipping flaky test

Tracking issue is #43255

dpoeschl committed:
Status indicator for whether added type name binds

sharwell committed:
Merge pull request #42733 from sharwell/polish-semicolon-completion

Polish Complete Statement (semicolon insertion)

JoeRobich committed:
Merge pull request #43246 from JoeRobich/enable-sdlvalidation

Enable SDL Validation in official builds

mizuniga committed:
Fixed the expected backslash in the Compiler Tests for Anonymous Types (#43211)

dpoeschl committed:
Fixup ImmutableArray usage in CreateCompensatingSignatureChange

jaredpar committed:
Merge pull request #43256 from jaredpar/flaky

Skipping flaky test

tmat committed:
Remove unnecessary feeds from NuGet.config (#43243)

dpoeschl committed:
Be sure to ignore added parameters when permuting

dpoeschl committed:
Fix some tests with bad baselines

dpoeschl committed:
Fix some C# doc comment tests

dpoeschl committed:
Fix null reference in C# doc comment management

jaredpar committed:
Update Language Feature Status

Reflecting our current status

chsienki committed:
Remove older source generators proposal, and redirect to the new one

CyrusNajmabadi committed:
Merge pull request #43083 from faso/master

CanAddNullCheck update for nullable reference types

dibarbet committed:
Remove previous docker container

In addition to killing the previous docker container, we should remove it so we can re-use the name.

dpoeschl committed:
Update VBSyntaxFacts.IsNamedParameter

Commit:
Merge pull request #43238 from jasonmalinowski/fix-richnav-privateassets

Mark RichCodeNav.EnvVarDump as private asset

Commit:
Merge pull request #43272 from dotnet/dibarbet-remove-container

Remove previous docker container

jmarolf committed:
add clarifying comment for when we can turn this on for dotnet builds

CyrusNajmabadi committed:
Merge remote-tracking branch 'upstream/master' into sourceAssembly

Commit:
Merge pull request #43081 from CyrusNajmabadi/sourceAssembly

Record the project associated with a source assembly.

dpoeschl committed:
Fix most C# trivia bugs

dpoeschl committed:
Include removed arguments when producing leading trivia for new arguments

dpoeschl committed:
Fix display of parameters with default value of "default"

dpoeschl committed:
Fix integration tests

dpoeschl committed:
Improve indicator in ChangeSignature dialog when a type doesn't bind

Fixes #41782

CyrusNajmabadi committed:
Support throw exprs in use-conditional

CyrusNajmabadi committed:
Add tests

CyrusNajmabadi committed:
Add support for return statements.

CyrusNajmabadi committed:
Share code.

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Move

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Move

dpoeschl committed:
Adjust ChangeSig Dialog size & datagrid column widths

dpoeschl committed:
AddParameterViewModelTests and minor ViewModel tweaks

CyrusNajmabadi committed:
Warning

dpoeschl committed:
Better AddParameter viewmodel separation

dpoeschl committed:
Setting kind to Required should undo injection kind Omit

dpoeschl committed:
Focus controls during integration tests

dpoeschl committed:
Fix databinding issue and workaround WPF bug

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Fix

mavasani committed:
Disable CodeStyle analyzers by default in CI for Roslyn.sln

Workaround for https://github.com/dotnet/roslyn/issues/42166

mavasani committed:
Explicitly enforce IDE0055 as a warning in .editorconfig

mavasani committed:
Address feedback - skip non-configurable diagnostics

tmat committed:
Simplify analyzer assembly loaders, analyzer reference serialization (#43009)

* Simplify analyzer assembly loaders, analyzer reference serialization

* Fold ReferenceSerializationService into SerializerService

* Add nullable annotations

* Compare analyzer ids instead of instances when calculating SkippedHostAnalyzersInfo

sharwell committed:
Use a full path for AnalyzerReference in AnalyzerRunner

dpoeschl committed:
PR Feedback

CyrusNajmabadi committed:
Move file

CyrusNajmabadi committed:
Move file

CyrusNajmabadi committed:
share code

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Simplify

CyrusNajmabadi committed:
Simplify

jmarolf committed:
Merge pull request #43233 from jmarolf/infra/reduce-restore-times

use static graph to reduce restore times

dpoeschl committed:
PR Feedback

dpoeschl committed:
Split into IsNamedArgument and IsNameOfNamedArgument

dpoeschl committed:
PR Feedback

mavasani committed:
Update eng/targets/Imports.targets

CyrusNajmabadi committed:
Merge pull request #43293 from CyrusNajmabadi/throwExprs

Support converting assignments/returns to conditional expressions, even when they contain a throw statement.

JoeRobich committed:
Merge remote-tracking branch 'origin/master' into merges/release/dev16.6-to-master

CyrusNajmabadi committed:
Fix sorting of 'wrapping' items.

dpoeschl committed:
Deduplicate the Permute/Add logic in each ChangeSignatureService

dpoeschl committed:
Deduplicate more code in VBChangeSigService

CyrusNajmabadi committed:
Add tests

dpoeschl committed:
Improve checks for DocCommentNames.ParameterElementName

mavasani committed:
Merge pull request #43178 from mavasani/PortNewAnalyzer

Port CSharpUseLocalFunction analyzer/tests to shared layer

dpoeschl committed:
Remove unused code

dpoeschl committed:
Update comment

dpoeschl committed:
Rid GetContentFromDocumentationCommentTriviaSyntax of ToArray calls

Commit:
Merge pull request #43317 from CyrusNajmabadi/fixWrappingSorting

Fix sorting of 'wrapping' items.

Commit:
Merge pull request #43239 from dotnet/merges/release/dev16.6-to-master

Merge release/dev16.6 to master

tmat committed:
Move host analyzer references to Solution snapshot (#42240)

* Add Solution.AnalyzerReferences

* HostDiagnosticAnalyzers

* Move types down to Workspaces

sharwell committed:
Merge pull request #43101 from sharwell/fix-analyzer-runner

Fix AnalyzerRunner failure to load Microsoft.Build.Framework

sharwell committed:
Merge remote-tracking branch 'dotnet/master' into regex-features

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20200413.4 (#43333)

- Microsoft.DotNet.Arcade.Sdk: 1.0.0-beta.20208.8 -> 1.0.0-beta.20213.4

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

jaredpar committed:
Update docs/Language Feature Status.md

sharwell committed:
Add a timeout for InteractiveWindow.SubmitText

jcouv committed:
Add link to test plan

dpoeschl committed:
Fix C# AddNameToArgument to support AttributeArguments

dpoeschl committed:
Add test for VB Attributes with omitted arguments

tmat committed:
Import Microsoft.VsSDK.targets conditionally

VSToolsPath might not be initialized during restore.

tmat committed:
Update Roslyn.VisualStudio.InteractiveComponents.csproj

dpoeschl committed:
Rename CompensatingParameterConfigurations and add doc comment

dpoeschl committed:
Dedupe some of IsParamsArrayExpanded

dpoeschl committed:
Remove unnecessary nullable !

dpoeschl committed:
Rename IsCallsiteError to IsCallsiteTodo and document Parameter properties

sharwell committed:
Merge pull request #43117 from sharwell/simplify-test

Simplify structure tests using new markup syntax

sharwell committed:
Document and validate arguments for RemoveOrTransformAll

sharwell committed:
Add a timeout for Editor.WaitForActiveView

sharwell committed:
Update Xunit.Combinatorial to 1.3.2

sharwell committed:
Remove IterationDataAttribute

This attribute has been superseded by CombinatorialRangeAttribute.

sharwell committed:
Fix NullReferenceException in FailFast handling

sharwell committed:
Avoid duplicate exception allocations on failure path

dpoeschl committed:
Use SetProperty from AbstractNotifyPropertyChanged

dpoeschl committed:
PR Feedback

dpoeschl committed:
Correctly resize AddParameter (the type input box grows)

dpoeschl committed:
Put tests in the right files.

dpoeschl committed:
Remove irrelevant comments

JoeRobich committed:
Merge pull request #43343 from dotnet/dev/tmat/ic-fix

Import Microsoft.VsSDK.targets conditionally

dpoeschl committed:
Cite the bug causing incorrect named args in attribute construction.

dpoeschl committed:
Assert that we have an IsParams parameter symbol when generating explicit array

mavasani committed:
Merge pull request #43303 from mavasani/DisableCodeStyleAnalyzersByDefaultInCI

Disable CodeStyle analyzers by default in all explicit builds of Roslyn.sln (i.e. csc/vbc invocations both inside Visual Studio and from command line prompt)

allisonchou committed:
Merge pull request #42914 from douglasg14b/patch-1

Adds syntax highlighting to code examples

dpoeschl committed:
Nullable enable the CSharpChangeSignatureServices

Therzok committed:
Use local instead of captured foreach variable

sharwell committed:
Merge pull request #43340 from sharwell/full-analyzer-path

Use a full path for AnalyzerReference in AnalyzerRunner

Therzok committed:
Remove local variable, it's only needed for value types

Commit:
Merge pull request #43268 from jaredpar/doc

Update Language Feature Status

Commit:
Merge pull request #43335 from dotnet/merges/release/dev16.6-to-master

Merge release/dev16.6 to master

sharwell committed:
Merge pull request #43357 from Therzok/patch-3

Use captured foreach variable

sharwell committed:
Merge pull request #43352 from sharwell/failfast-nre

Fix NullReferenceException in FailFast handling

sharwell committed:
Capture heap dump when integration test crashes in CI

Commit:
Merge pull request #43269 from chsienki/deprecate_old_generator_docs

Remove older source generators proposal, and redirect to the new one

sharwell committed:
Gate the construction of the singleton RemoteWorkspace

sharwell committed:
Make sure IPersistentStorage instances are disposed

dpoeschl committed:
Update databindings immediately instead of on focus lost

dpoeschl committed:
Misc. data consistency measures

dpoeschl committed:
Show warning in ChangeSig dialog for duplicated param names

No warning in AddParam dialog yet

dpoeschl committed:
ChangeSignature - Allow removal of added optional parameters

dpoeschl committed:
Fix broken unit test baseline

sharwell committed:
Merge pull request #43370 from sharwell/remote-workspace

Gate the construction of the singleton RemoteWorkspace

sharwell committed:
Merge pull request #43364 from sharwell/capture-dump

Capture heap dump when integration test crashes in CI

sharwell committed:
Merge pull request #43353 from sharwell/update-combinatorial

Update Xunit.Combinatorial

sharwell committed:
Merge pull request #34802 from sharwell/regex-features

🧹 Move Regex embedded language to the Features layer

dpoeschl committed:
Handle OmittedArgumentSyntax when adding a name to a VB argument

sharwell committed:
Merge pull request #43277 from sharwell/nullable-formatter

Enable nullable reference types in the formatter

sharwell committed:
Merge pull request #43338 from sharwell/submittext-timeout

Add a timeout for InteractiveWindow.SubmitText

dpoeschl committed:
PR  Feedback

dpoeschl committed:
Merge remote-tracking branch 'dotnet/master' into AddOptionalParameter

dpoeschl committed:
Update type name status strings

dpoeschl committed:
Misc cleanup to ParameterViewModels

dpoeschl committed:
Group AddParameter controls, add some spacing, adjust strings

dpoeschl committed:
Fix formatting
